WordPress/gutenberg
# Table Block - Ensure empty classname is not output onto table element

## Description
The current version of the table block in master, when saved produces html with an unnecessary class attribute on the table element:
```html

<figure class="wp-block-table"><table class=""><tbody><tr><td></td><td></td></tr></tbody></table></figure>

```

This PR removes that when the class is an empty string. Questionable whether this is even really a bug, but I think it's good to keep the HTML as clean as possible.

Initially I thought this change would require a deprecation, but that doesn't seem to be the case. Following the testing steps confirms that.



## How has this been tested?
1. Start a new post
2. Switch to the Code Editor
3. Paste the following table block HTML (which represents how the block used to render):
```html

<figure class="wp-block-table"><table class=""><tbody><tr><td></td><td></td></tr></tbody></table></figure>

```
4. Switch to the Visual Editor, confirm that the block has no validation errors
5. Switch back to the Code Editor (or preview the post and inspect the HTML)
6. Observe that the unnecessary class name is no longer produced

## Types of changes

Bug fix (non-breaking change which fixes an issue)



## Checklist:
